### PR TITLE
Tensorflow plugin implementation (from PR #141)

### PR DESCRIPTION
**Pull request by wild-endeavor**
Please see original PR https://github.com/lyft/flytekit/pull/141

Just bumped the version, and cleaned up some python2 stuff, and ran `make fmt`

### CODE REVIEW

**Comment by wild-endeavor** [comment]
cc @pingsutw

File: flytekit/sdk/tasks.py [hunk 1]
```diff
@@ -1336,3 +1337,167 @@ def wrapper(fn):
...
+        inject bespoke logic into the base Flyte programming model.
+
+    :rtype: flytekit.common.tasks.sdk_runnable.SdkRunnableTask
+    """
+
+    def wrapper(fn):
+        return (cls or _sdk_tensorflow_tasks.SdkTensorFlowTask)(
+            task_function=fn,
+            task_type=_common_constants.SdkTaskType.TENSORFLOW_TASK,
+            discovery_version=cache_version,
```
Comment (EngHabu) [inline flytekit/sdk/tasks.py:1480]:
```suggestion
            cache_version=cache_version,
```

File: flytekit/sdk/tasks.py [hunk 2]
```diff
@@ -1336,3 +1337,167 @@ def wrapper(fn):
...
+            retries=retries,
+            interruptible=interruptible,
+            deprecated=deprecated,
+            discoverable=cache,
```
Comment (EngHabu) [inline flytekit/sdk/tasks.py:1484]:
```suggestion
            cache=cache,
```

File: flytekit/common/tasks/tensorflow_task.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+class SdkTensorFlowTask(_sdk_runnable.SdkRunnableTask):
+    def __init__(
+        self,
+        task_function,
+        task_type,
+        discovery_version,
+        retries,
+        interruptible,
+        deprecated,
+        discoverable,
```
Comment (EngHabu) [inline flytekit/common/tasks/tensorflow_task.py:26]:
```suggestion
        cacheable,
```

File: flytekit/common/tasks/tensorflow_task.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+        """
+        return self._args
+
+
+class SdkTensorFlowTask(_sdk_runnable.SdkRunnableTask):
+    def __init__(
+        self,
+        task_function,
+        task_type,
+        discovery_version,
```
Comment (EngHabu) [inline flytekit/common/tasks/tensorflow_task.py:22]:
```suggestion
        cache_version,
```

File: flytekit/common/tasks/tensorflow_task.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        per_replica_memory_limit,
+        environment,
+    ):
+        tensorflow_job = _task_models.TensorFlowJob(
+            workers_count=workers_count, ps_replicas_count=ps_replicas_count, chief_replicas_count=chief_replicas_count
+        ).to_flyte_idl()
+        super(SdkTensorFlowTask, self).__init__(
+            task_function=task_function,
+            task_type=task_type,
+            discovery_version=discovery_version,
```
Comment (EngHabu) [inline flytekit/common/tasks/tensorflow_task.py:47]:
```suggestion
            discovery_version=cache_version,
```

File: flytekit/common/tasks/tensorflow_task.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+            deprecated=deprecated,
+            storage_request=per_replica_storage_request,
+            cpu_request=per_replica_cpu_request,
+            gpu_request=per_replica_gpu_request,
+            memory_request=per_replica_memory_request,
+            storage_limit=per_replica_storage_limit,
+            cpu_limit=per_replica_cpu_limit,
+            gpu_limit=per_replica_gpu_limit,
+            memory_limit=per_replica_memory_limit,
+            discoverable=discoverable,
```
Comment (EngHabu) [inline flytekit/common/tasks/tensorflow_task.py:59]:
```suggestion
            discoverable=cache,
```

**Review by EngHabu — DISMISSED** [review]

**Review by EngHabu — DISMISSED** [review]

File: flytekit/common/tasks/tensorflow_task.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+class SdkTensorFlowTask(_sdk_runnable.SdkRunnableTask):
+    def __init__(
+        self,
+        task_function,
+        task_type,
+        cache_version,
+        retries,
+        interruptible,
+        deprecated,
+        cacheable,
```
Comment (wild-endeavor) [inline flytekit/common/tasks/tensorflow_task.py:26]:
```suggestion
        cache,
```

**Review by EngHabu — APPROVED** [review]

**Pull request merged**